From d6aac5857a7aea11a7fc95926d587ecc824b6152 Mon Sep 17 00:00:00 2001
From: Cropi <alakatos@redhat.com>
Date: Wed, 2 Apr 2025 11:04:37 +0200
Subject: [PATCH] Update end of event detection

---
 auparse/auparse.c  | 5 ++++-
 src/ausearch-lol.c | 6 +++++-
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/auparse/auparse.c b/auparse/auparse.c
index c3e1fb9e..1b0c5a39 100644
--- a/auparse/auparse.c
+++ b/auparse/auparse.c
@@ -319,7 +319,9 @@ static void au_check_events(auparse_state_t *au, time_t sec)
                         } else if ( // FIXME: Check this v remains true
 				r->type == AUDIT_PROCTITLE ||
 				r->type == AUDIT_EOE ||
-				r->type < AUDIT_FIRST_EVENT ||
+				(r->type > AUDIT_LOGIN &&
+					r->type < AUDIT_FIRST_EVENT) ||
+				r->type == AUDIT_USER ||
 				r->type >= AUDIT_FIRST_ANOM_MSG ||
 				r->type == AUDIT_KERNEL ||
 				(r->type >= AUDIT_MAC_UNLBL_ALLOW &&
@@ -332,6 +334,7 @@ static void au_check_events(auparse_state_t *au, time_t sec)
         }
 }
 
+
 /*
  * au_terminate_all_events - Mark all events in 'BUILD' state to be COMPLETE
  *
diff --git a/src/ausearch-lol.c b/src/ausearch-lol.c
index 31c5ff2e..e2a6017d 100644
--- a/src/ausearch-lol.c
+++ b/src/ausearch-lol.c
@@ -259,7 +259,10 @@ static void check_events(lol *lo, time_t sec)
 				cur->status = L_COMPLETE;
 				ready++;
 			} else if (cur->l->e.type == AUDIT_PROCTITLE ||
-				    cur->l->e.type < AUDIT_FIRST_EVENT ||
+				    cur->l->e.type == AUDIT_EOE ||
+				    (cur->l->e.type > AUDIT_LOGIN &&
+				        cur->l->e.type < AUDIT_FIRST_EVENT) ||
+				    cur->l->e.type == AUDIT_USER ||
 				    cur->l->e.type >= AUDIT_FIRST_ANOM_MSG ||
 				    cur->l->e.type == AUDIT_KERNEL ||
 				    (cur->l->e.type >= AUDIT_MAC_UNLBL_ALLOW &&
@@ -272,6 +275,7 @@ static void check_events(lol *lo, time_t sec)
 	}
 }
 
+
 // This function adds a new record to an existing linked list
 // or creates a new one if its a new event
 int lol_add_record(lol *lo, char *buff)
-- 
2.49.0

