diff --git a/init.d/auditd.service b/init.d/auditd.service
index dd7ec694b..d5139ae92 100644
--- a/init.d/auditd.service
+++ b/init.d/auditd.service
@@ -6,6 +6,9 @@ DefaultDependencies=no
 ## uncomment the second so that network-online.target is part of After.
 ## then comment the first Before and uncomment the second Before to remove
 ## sysinit.target from "Before".
+## If using remote logging, ensure that the systemd-update-utmp.service file
+## is updated to remove the After=auditd.service directive to prevent a
+## boot-time ordering cycle.
 After=local-fs.target systemd-tmpfiles-setup.service
 ##After=network-online.target local-fs.target systemd-tmpfiles-setup.service
 Before=sysinit.target shutdown.target
